lib/RWMutex.ts

```diff
@@ -39,14 +39,25 @@ export default class RWMutex {
 
   /*
    * Acquires the write lock.
+   * @param {callback} - Optional callback
    * @return {Promise} - Promise that resolves when the lock is acquired, rejects if an error occurs
    */
-  async lock() {
-    // provides readable error messages, no need to catch and rethrow
-    const lock = await this._findOrCreateLock();
+  async lock(cb) {
+    let lock;
+    try {
+      lock = await this._findOrCreateLock();
+    } catch (err) {
+      if (typeof cb === "function") {
```

Comment:
I'm generally hesitant to add dependencies to libaries, but I agree that this is very cumbersome. I'll experiment with (1) and see if we can use callbackify in titanium-transactions so we don't have to solve this problem.